### ISSUE TITLE
Missing -Wreturn-local-addr warning with std::move

### ISSUE THREAD

**Issue by efriedma-quic** [llvm/llvm-project#47866]
|  |  |
| --- | --- |
| Bugzilla Link | [48522](https://llvm.org/bz48522) |
| Version | unspecified |
| OS | Windows NT |
| CC | @AaronBallman,@dwblaikie,@nickdesaulniers,@zygoloid |

## Extended Description 
Testcase:
```
#include <utility>
const int &byref() {
    int x = 0;
    return std::move(x);
}
```
This is returning the address of a local variable, but clang doesn't detect it.  gcc does detect this.

There are a couple ways we could address this:

1. Special-case calls to the symbol std::move.
2. Analyze the loop body of inline functions for a return statement that returns a reference argument/pointer (sort of like the LLVM IR "returned" attribute).

The static analyzer's core.StackAddressEscape analysis does correctly warn.  Not sure how far we want to go in the compiler, but this particular case seems low-cost/high-impact.

**Comment by xgupta** [comment]
Seems this issue is fixed on main - https://godbolt.org/z/ET577WErc.

clang warns with - 
<source>:4:22: warning: reference to stack memory associated with local variable 'x' returned [-Wreturn-stack-address]
    return std::move(x);
                     ^
1 warning generated.
Compiler returned: 0

**Comment by xgupta** [comment]
> Seems this issue is fixed on main - https://godbolt.org/z/ET577WErc.
> 
> clang warns with -
> 
> :4:22: warning: reference to stack memory associated with local variable 'x' returned [-Wreturn-stack-address] return std::move(x); ^ 1 warning generated. Compiler returned: 0

We can close it then.
